maestrowf/utils.py

```diff
@@ -349,3 +350,95 @@ def map_level(log_lvl):
+
+    def __init__(self, meta_parameters_path: str, output_path: str):
+        self.meta_parameters_path: str = meta_parameters_path
+        self.output_path: str = output_path
+
+        self.formatted_labels: list = []
+        self.formatted_values: list = []
+        self.formatted_names: list = []
+
+    def construct_parameters(self):
+        """Gather raw parameters, build a formatted block, output to a file."""
+        formatted_items: tuple(list, list, list) = self.gather_parameters()
+        global_parameters_string: str = self.build_parameters(formatted_items)
+        self.dump_parameters(global_parameters_string)
+
+    def gather_parameters(self):
+        """Gathers the raw names, values and labels from a parameters.yaml and
+        builds indiviudal lists for those properties."""
+        raw_parameters: dict = {}
+
+        with open(self.meta_parameters_path) as f:
+            raw_parameters: object = yaml.load(f, Loader=yaml.Loader)
+
+        for raw_values in raw_parameters.values():
+            for i in range(len(list(raw_values.items())[0][1].items())):
+                labels_list: list = list(raw_values.items())[0][1]
+                label: str = list(labels_list.items())[i][1]
+
+                self.formatted_labels.append(label)
+
+                values_list: list = list(raw_values.items())[1][1]
+                value: str = list(values_list.items())[i][1]
+
+                names_list: list = list(raw_values.items())[1][1]
+                name: str = list(names_list.items())[i][0]
+                name: str = name.replace('$(', '').replace(')', '')
+
+                self.formatted_names.append(
+                    self.formatted_labels[i].replace('.1', ''))
+
+                if type(value) is list:
+                    value: str = '\n\t\t' + "- {}".format(str(value))
+                else:
+                    value: str = '[' + str(value) + ']'
+
+                self.formatted_values.append(value)
+
+        return (self.formatted_names,
+                self.formatted_values,
+                self.formatted_labels)
+
+    def build_parameters(self, formatted_items: tuple(list, list, list)):
+        """Builds a global.parameters block given formatted names, values, and labels
+        then saves to generated_global_parameters.yaml.
+
+        Example:
+            global.parameters:
+                ANTIGEN_CHAINS_IN_STRUCT:
+                    values: [A]
+                    label: [ANTIGEN_CHAINS_IN_STRUCT.1]
+        """
+        global_parameters_block: list = []
+        global_parameters_block.append("global.parameters:\n")
+
+        names: list = formatted_items[0]
+        values: list = formatted_items[1]
+        labels: list = formatted_items[2]
+
+        for i in range(len(values)):
+            blocks: dict = {
+                'name': names[i],
+                'value': values[i],
+                'label': labels[i]
+            }
+            formatted_block: str = \
```

Comment:
I think that we discussed this in a 1:1 a bit ago before I had to take some leave, but it looks like you've manually formatted a YAML line here. It might be worth compiling this into a directory structure that matches the specification, then use a markup language specific library to dump it to its appropriate format.